### ISSUE TITLE
Add strict-optional configuration option (or separate rule) for T | None argument mismatches

### ISSUE THREAD

**Issue by FroguelYipit** [astral-sh/ty#2421]
### Summary

This is more of a feature request, but since I didn't see that option, I'm opening as a bug. 

When using third-party libraries with incomplete type hints, ty frequently flags `invalid-argument-type` errors when passing `T | None` to functions that accept `T`. This happens because the library's type stubs incorrectly annotate a parameter as `T` when it actually accepts `None` at runtime. 

Example
```py
def lib_func(a: int = None) -> int:
    return a or 1

a = None
lib_func(a=a) 
```
then `ty` will flag the `a=a` as being wrong, despite there being no error at runtime. I understand this might be desired behavior in some cases, which is why I'm asking for a config to switch on or off. My issue is specific for this `None` that work at runtime. Ideally they'd be warnings not errors, but only current option would be to disable all invalid argument checks which is not an option. 

### Version

ty 0.0.10 (d18902cdc 2026-01-07)

**Comment by carljm** [comment]
Hmm, I was hoping the Python ecosystem by this point had mostly moved on from relying on mypy's old non-strict-optional behavior, but it sounds like at least some libraries have not?

I don't love the idea of adding an option for this -- IMO the annotations in the library are just wrong and should be fixed. But we can consider it, and see if there's high demand for it.

Do any type checkers other than mypy offer a similar option?
